index.js

```diff
@@ -26,4 +26,11 @@ isStream.transform = stream =>
 	typeof stream._transform === 'function' &&
 	typeof stream._transformState === 'object';
 
+isStream.type = stream =>
+	isStream.transform(stream) ? 'transform' :
+		isStream.duplex(stream) ? 'duplex' :
+			isStream.writable(stream) ? 'writable' :
+				isStream.readable(stream) ? 'readable' :
+					isStream(stream) ? 'unknown' : 'undefined';
```

Comment:
Both these cases should be the undefined value (NOT string)

Reply:
In this case we can refactoring the sentence like this:
```js
	isStream.transform(stream) ? 'transform' :
		isStream.duplex(stream) ? 'duplex' :
			isStream.writable(stream) ? 'writable' :
				isStream.readable(stream) ? 'readable' : undefined;
```
What do you think about this approach?

Reply:
I think it would be more readable as just if-statements.

Reply:
Done ccbe470.

Reply:
https://github.com/sindresorhus/is-stream/pull/14#discussion_r465192065